### PR TITLE
Store endpoint

### LINKED ISSUE #44: Add Store resource
## Issue
- [ ] Style (non-breaking style code)
- [ ] Testing (add testing to code)
- [ ] Bug fix (change which fixes an issue)
- [ ] Refactor (improvement to the code)
- [x] New feature (change which adds functionality)
- [ ] Chore
## Issue Summary

Add Store resource with get method

## Resources

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [x] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [ ] Bug fix or refactor (change which fixes an issue)
- [x] New feature (change which adds functionality)

## Summary of the change.

- added the `store` resource (with `get` method) and model
- changed `post` method of `atriskuser` resource to automatically search for the closest `store`   
- added a one-to-one relationship `atriskuser`/`store`

This closes #44 #43 

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -12,10 +12,14 @@ Flask-SQLAlchemy ="*"
 Flask-Migrate="*"
 Flask-Script="*"
 psycopg2="*"
-
+requests = "^2.23.0"
+ipdb = "^0.13.2"
```
Comment (adumortier) [inline pyproject.toml:16]: debugger equivalent of `pry`

File: pyproject.toml [hunk 2]
```diff
@@ -12,10 +12,14 @@ Flask-SQLAlchemy ="*"
...
+gunicorn = "^20.0.4"
```
Comment (adumortier) [inline pyproject.toml:17]: Python WSGI HTTP Server

File: trackbasket_be/app.py [hunk 3]
```diff
@@ -17,13 +21,16 @@
...
+
+
 app.config['SQLALCHEMY_DATABASE_URI'] = 'postgresql://%(user)s:\%(pw)s@%(host)s:%(port)s/%(db)s' % POSTGRES
 app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
 
 db.init_app(app)
 
 api.add_resource(Volunteers, '/volunteer/<string:id>')
 api.add_resource(AtRiskUsers, '/atriskuser/<string:id>')
+api.add_resource(Stores, '/store/<string:id>')
```
Comment (adumortier) [inline trackbasket_be/app.py:33]: adds the `store` resource to the app

File: trackbasket_be/app.py [hunk 4]
```diff
@@ -1,10 +1,14 @@
+
 from flask import Flask, request
 from flask_restful import Resource, Api
 from flask_sqlalchemy import SQLAlchemy
-from models.volunteer import db, Volunteer
+from models.basemodel import db
+from models.volunteer import Volunteer
 from resources.volunteer import Volunteers
-from models.at_risk_user import db, AtRiskUser
+from resources.store import Stores
```
Comment (adumortier) [inline trackbasket_be/app.py:8]: removed duplicated imports of `db`

File: trackbasket_be/models/at_risk_user.py [hunk 5]
```diff
@@ -11,13 +14,14 @@ class AtRiskUser(BaseModel, db.Model):
   state = db.Column(db.String)
   zipcode = db.Column(db.String)
   phone_number = db.Column(db.String)
+  store = db.relationship('Store', backref=backref('at_risk_user', uselist=False))
```
Comment (adumortier) [inline trackbasket_be/models/at_risk_user.py:17]: creates the one-to-one relationship between at `at_risk_user` and the closest `store` (`uselist` flag would be set to true for a one-to-many relationship)

File: trackbasket_be/resources/store.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+from models.basemodel import BaseModel
+from models.krogerservice import Krogerservice
+import requests
+
+class Stores(Resource):
+
+  def get(self,id):
+    request_data = request.json
+    zipcode = request_data['zipcode']
+    return Krogerservice.closest_store(zipcode)
```
Comment (adumortier) [inline trackbasket_be/resources/store.py:11]: refactored the `store` resource to move the work done in the model

File: trackbasket_be/models/store.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  name = db.Column(db.String)
+  location_id = db.Column(db.String)
+  name = db.Column(db.String)
+  address = db.Column(db.String)
+  city = db.Column(db.String)
+  state = db.Column(db.String)
+  zipcode = db.Column(db.String)
+  latitude = db.Column(db.String)
+  longitude = db.Column(db.String)
+  at_risk_user_id = db.Column(db.Integer, db.ForeignKey('at_risk_users.id'))
```
Comment (adumortier) [inline trackbasket_be/models/store.py:16]: add the `at_risk_user_id` as a foreign key to the `stores` table

File: trackbasket_be/models/krogerservice.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    headers = {'Content-Type': 'application/x-www-form-urlencoded', 'Authorization': authorization}
+    response = requests.post('https://api.kroger.com/v1/connect/oauth2/token', headers=headers, data=data).json()
+    global access_token 
+    access_token = response['access_token']
+    return access_token
+  
+  @classmethod
+  def closest_store(cls, zipcode):
+    parameters = {'filter.zipCode.near': zipcode, 'filter.limit':1}
+    headers = {'Authorization': 'Bearer {}'.format(Krogerservice.return_token())}
```
Comment (adumortier) [inline trackbasket_be/models/krogerservice.py:30]: Token is used directly if one is available. If it does not exist or if it has expired, we make a new Kroger API request

File: trackbasket_be/models/krogerservice.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+      return access_token
+
+  @classmethod
+  def refresh_token(cls):
+    client = os.environ.get("CLIENT")
+    authorization = 'Basic {}'.format(client)
+    data = { 'grant_type':'client_credentials', 'scope': 'product.compact' }
+    headers = {'Content-Type': 'application/x-www-form-urlencoded', 'Authorization': authorization}
+    response = requests.post('https://api.kroger.com/v1/connect/oauth2/token', headers=headers, data=data).json()
+    global access_token
```
Comment (mronauli) [inline trackbasket_be/models/krogerservice.py:23]: also super clutch

File: trackbasket_be/models/krogerservice.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    try:
+      access_token
+    except NameError:
+      return Krogerservice.refresh_token()
+    else:
+      return access_token
+
+  @classmethod
+  def refresh_token(cls):
+    client = os.environ.get("CLIENT")
```
Comment (mronauli) [inline trackbasket_be/models/krogerservice.py:18]: do you think this will impact heroku deployment?

**Pull request merged**